### PR TITLE
LS-48825 Document the inferred_service_rule resource

### PR DESCRIPTION
**Pull request by gdfast**
What does this PR do?
---------------------

This PR adds provider documentation for the new inferred_service_rule resource. Specifically, it
- adds a doc template file for the resource
- updates some descriptions in the resource Schema to better accomodate the auto-generated doc language
- adds the markdown doc, generated by `tfplugindocs`

How did you test this?
---------------------

- Ran tfplugindocs and inspected the rendered
- Copy-pasted the markdown doc into

### CODE REVIEW

File: lightstep/resource_inferred_service_rule.go [hunk 1]
```diff
@@ -39,7 +38,7 @@ func getResourceInferredServiceRuleSchema() map[string]*schema.Schema {
 			ValidateFunc: validation.StringLenBetween(0, 190),
 		},
 		"description": {
-			Description:  "An optional description to describe the rule and what services it should infer",
+			Description:  "A description of the rule and what services it should infer",
```
Comment (gdfast) [inline lightstep/resource_inferred_service_rule.go:41]: This was added to the "Optional" section in the docs, so no need to write that it's optional

File: lightstep/resource_inferred_service_rule.go [hunk 2]
```diff
@@ -54,7 +53,7 @@ func getResourceInferredServiceRuleSchema() map[string]*schema.Schema {
 			MinItems: 1,
 		},
 		"group_by_keys": {
-			Description: "A list of attribute keys whose values will be included in the inferred service name",
+			Description: "Attribute keys whose values will be included in the inferred service name",
```
Comment (gdfast) [inline lightstep/resource_inferred_service_rule.go:56]: tfplugindocs automatically adds the types to the doc, so no need to write that this is a list

File: docs/resources/inferred_service_rule.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "lightstep_inferred_service_rule Resource - terraform-provider-lightstep"
+subcategory: ""
+description: |-
+
+---
+
+# lightstep_inferred_service_rule (Resource)
+
+~> 🚧 This resource is under development and is not generally available yet. 🚧
```
Comment (adil-sultan) [inline docs/resources/inferred_service_rule.md:11]: Did this get left in here?

**Review by adil-sultan — APPROVED** [review]
Just one comment but LGTM!

**Review by carolynblumberg — APPROVED** [review]

File: docs/resources/inferred_service_rule.md (hunk 3 above)
Comment (gdfast) [reply docs/resources/inferred_service_rule.md:11]: yeah intentionally! The Public API isn't ready for use yet, so this resource won't work yet. We can remove this line once we Go Live

I was inspired by a similar line on the user binding resource

**Pull request merged**